frontendGraphImpl/src/main/java/com/virtuslab/gitmachete/frontend/graph/impl/repository/RepositoryGraph.java

```diff
@@ -1,100 +1,78 @@
 
-public class RepositoryGraph {
-  @Getter
-  @SuppressWarnings("ConstantName")
-  private static final RepositoryGraph nullRepositoryGraph = new RepositoryGraphBuilder()
-      .repository(NullRepository.getInstance()).build();
+public class RepositoryGraph implements IRepositoryGraph {
 
-  private final List<IGraphNode> nodes;
+  private final List<IGraphItem> items;
   private final List<List<Integer>> positionsOfVisibleEdges;
-
-  private final PrintElementGeneratorImpl printElementGenerator;
+  private final PrintElementGenerator printElementGenerator;
 
   @SuppressWarnings({"nullness:argument.type.incompatible", "nullness:assignment.type.incompatible"})
-  public RepositoryGraph(List<IGraphNode> nodes, List<List<Integer>> positionsOfVisibleEdges) {
-    this.nodes = nodes;
+  public RepositoryGraph(List<IGraphItem> items, List<List<Integer>> positionsOfVisibleEdges) {
+    this.items = items;
     this.positionsOfVisibleEdges = positionsOfVisibleEdges;
-
-    GraphElementManager graphElementManager = new GraphElementManager(/* repositoryGraph */ this);
-    printElementGenerator = new PrintElementGeneratorImpl(/* graph */ this, graphElementManager);
+    this.printElementGenerator = new PrintElementGenerator(/* graph */ this);
   }
 
-  public Collection<? extends PrintElement> getPrintElements(@NonNegative int rowIndex) {
-    return printElementGenerator.getPrintElements(rowIndex);
+  public Collection<? extends IPrintElement> getPrintElements(@NonNegative int nodeIndex) {
+    return printElementGenerator.getPrintElements(nodeIndex);
   }
 
   @SuppressWarnings("upperbound:argument.type.incompatible")
-  public IGraphNode getGraphNode(@NonNegative int rowIndex) {
-    return nodes.get(rowIndex);
+  public IGraphItem getGraphItem(@NonNegative int itemIndex) {
+    return items.get(itemIndex);
   }
 
   @NonNegative
   public int nodesCount() {
-    return nodes.size();
-  }
-
-  public int getNodeId(int nodeIndex) {
-    assert nodeIndex >= 0 && nodeIndex < nodesCount() : "Bad nodeIndex: " + nodeIndex;
-    return nodeIndex;
+    return items.size();
   }
 
   /**
    * Adjacent edges are the edges that are visible in a row and directly connected to the node (branch/commit node)
    * of this row. See {@link RepositoryGraph#getVisibleEdgesWithPositions} for more details.
    *
    * @param nodeIndex node index
-   * @param filter edge filter
    * @return list of adjacent edges in a given node index
    */
-  public java.util.List<GraphEdge> getAdjacentEdges(@NonNegative int nodeIndex, EdgeFilter filter) {
-    if (filter == EdgeFilter.SPECIAL) {
-      return Collections.emptyList();
-    }
-
+  public java.util.List<GraphEdge> getAdjacentEdges(@NonNegative int nodeIndex) {
     java.util.List<GraphEdge> adjacentEdges = new SmartList<>();
     @SuppressWarnings("upperbound:argument.type.incompatible")
-    IGraphNode currentNode = nodes.get(nodeIndex);
+    IGraphItem currentItem = items.get(nodeIndex);
```

Comment:
Hmm is it okay to get `nodeIndex` from `items`? Shouldn't it be `itemIndex` then?

Reply:
Btw this stuff qualifies for https://checkerframework.org/manual/#fenum-checker ;) but that's ofc beyond the scope of this PR

Reply:
As in, `@Fenum("itemIndex")` vs `@Fenum("nodeIndex")`

Reply:
Or actually... are the those two index domain equal? As in, a correct node index is always a correct item index and vice versa? :thinking:

Reply:
If so, then no extra checker would be needed

Reply:
> Or actually... are the those two index domain equal? As in, a correct node index is always a correct item index and vice versa?

They indeed do equal!

However, some renames and argument comments are needed here